pkg/scheduler/dependency_graph.go

```diff
@@ -83,6 +92,35 @@ func newScheduledResourceFor(r interfaces.Resource, context *GraphContext) *Sche
 	}
 }
 
+type DependencyList client.DependencyList
+
+func (d *DependencyList) Len() int {
+	return len(d.Items)
+}
+
+func (d *DependencyList) Less(i, j int) bool {
+	if d.Items[i].CreationTimestamp.Equal(d.Items[j].CreationTimestamp) {
+		return d.Items[i].UID < d.Items[j].UID
+	}
+	return d.Items[i].CreationTimestamp.Before(d.Items[j].CreationTimestamp)
+}
+
+func (d *DependencyList) Swap(i, j int) {
```

Comment:
Thanks for clarification. Please consider adding these lines:
```
var _ sort.Interface = &DependencyList{}
var _ sort.Interface = &ReplicaList{}
```
This way we can make sure that these lists actually implement the interface on compile time.